tools/test/LcovMerger/javatests/com/google/devtools/lcovmerger/SourceFileCoverageTest.java

```diff
@@ -77,4 +78,16 @@ public void testMerge() {
         SourceFileCoverage.merge(sourceFile1, sourceFile2), linesExecution1, linesExecution2);
   }
 
+  @Test
+  public void testConstructorWithVirtualIncludeFile() {
+    assertThat(new SourceFileCoverage(
+        "bazel-out/k8-fastbuild/bin/include/common/_virtual_includes/strategy/strategy.h")
```

Comment:
Can you try with a target:

cc_library(
  name = "asdf"
  hdrs = [ "foo/bar/baz/asdf.h" ], // header is in a subdirectory
  strip_include_prefix = "foo",
  include_prefix = "yo/lo"
)

I'd just go crazy and test all the possible variants :)